src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php

```diff
@@ -165,6 +183,15 @@ public function collect(Request $request, Response $response, \Exception $except
                     'user_checker' => $firewallConfig->getUserChecker(),
                     'listeners' => $firewallConfig->getListeners(),
                 );
+
+                // generate exit impersonation path from current request
+                if ($this->data['impersonated'] && null !== $switchUserConfig = $firewallConfig->getSwitchUser()) {
+                    $exitPath = $request->getRequestUri();
+                    $exitPath .= null === $request->getQueryString() ? '?' : '&';
+                    $exitPath .= urlencode($switchUserConfig['parameter']).'=_exit';
```

Comment:
We should add/use a `const EXIT = '_exit'` into `SwitchUserListener` to avoid relying on "magic strings"? It's worth?